.github/workflows/build-test.yaml

```diff
@@ -9,9 +9,30 @@ permissions:
   contents: read # This is required for actions/checkout
 
 jobs:
+  compute-tag:
+    runs-on: ubuntu-latest
+    outputs:
+      tag: ${{ steps.get_tag.outputs.TAG }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Get the latest tag
+        id: get_tag
+        run: echo "TAG=v0.0.1-test" >> "$GITHUB_OUTPUT"
+      - name: Echo the tag
+        run: echo ${{ steps.get_tag.outputs.TAG }}
   build-image:
-    uses: kyma-project/test-infra/.github/workflows/image-builder.yml@main
+    needs: compute-tag
+    uses: kyma-project/test-infra/.github/workflows/image-builder.yml@${{ github.event.pull_request_target.head.ref }}
```

Comment:
It must be: uses: kyma-project/test-infra/.github/workflows/image-builder.yml@main